nix/module.nix

```diff
@@ -0,0 +1,114 @@
+{ options, config, lib, pkgs, ... }:
+
+with lib;
+with lib.types;
+let
+  cfg = config.programs.fw-fanctrl;
+  fw-ectool = pkgs.callPackage ./packages/fw-ectool.nix {};
+  fw-fanctrl = pkgs.callPackage ./packages/fw-fanctrl.nix {};
+  defaultConfig = builtins.fromJSON (builtins.readFile ../config.json);
+in
+{
+  options.programs.fw-fanctrl = {
+    enable = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Enable fw-fanctrl systemd service and install the needed packages.
+      '';
+    };
+    config = {
+      defaultStrategy = mkOption {
+        type = str;
+        default = defaultConfig.defaultStrategy;
+        description = "Default strategy to use";
+      };
+      strategyOnDischarging = mkOption {
+        type = str;
+        default = defaultConfig.strategyOnDischarging; 
+        description = "Default strategy on discharging";
+      };
+      batteryChargingStatusPath = mkOption {
+        type = str;
+        default = defaultConfig.batteryChargingStatusPath;
+        description = "";
+      };
+      strategies = mkOption {
+        default = defaultConfig.strategies;
+        type = attrsOf (submodule (
+          { options, name, ... }:
+          {
+            options = {
+              name = mkOption {
+                type = str;
+                default = "";
+                description = "Name of the strategy";
+              };
+              fanSpeedUpdateFrequency = mkOption {
+                type = int;
+                default = 5;
+                description = "How often the fan speed should be updated in seconds";
+              };
+              movingAverageInterval = mkOption {
+                type = int;
+                default = 25;
+                description = "";
```

Comment:
The program will take the x last temperatures (1 per second), do the average, compare it to the actual temp and take the lowest one

Reply:
this option defines the number of lest temps to take (defaults to 30)